### ISSUE TITLE
Support hot-update client TLS certificate and CA

### ISSUE THREAD

**Issue by ukcuf** [haproxy/haproxy#427]
## Output of `haproxy -vv` and `uname -a`

```
global
    log stdout local0
    stats socket /tmp/admin-1.sock level admin process 1 mode 0660 expose-fd listeners
    tune.ssl.default-dh-param 2048

defaults
    log global
    maxconn 3000
    mode http
    option httplog
    option http-use-htx
    option redispatch
    option logasap

frontend myproxy
    bind :58080
    default_backend mybackend

backend mybackend
    server localserver 127.0.0.1:8443 ssl verify required ca-file /tmp/pem/ca.crt crt /tmp/pem/tls.pem force-tlsv13
```

## What should haproxy do differently? Which functionality do you think we should add?

haproxy 2.1 has feature to hot-update server TLS certificate(contains private key) through admin socket management command. However, the command does not support to update client TLS certificate and CA. See below output when we try the command with haproxy configuration from above section:

```bash
# echo -e "set ssl cert /tmp/pem/tls.pem <<\n$(cat /tmp/pem/tls.pem)\n" \
       | socat /tmp/admin-1.sock -
Can't replace a certificate which is not referenced by the configuration!
Can't update /tmp/pem/tls.pem
```

It will be super great if: 
1. command `set ssl cert`,`abort ssl cert` and `commit ssl cert` can also support to hot-update client certificate which is used in `server` directive in the backend section. 
2. a new command to hot-update CA used with `ca-file`, for `bind options`, `Server and default-server options`)

## What are you trying to do?

We have a series of Kubernetes deployments running on Kubernetes clusters. As service mesh data plane implementation, all internal traffic between pods are managed by haproxy sidecar running in each pod. So all internal traffic across different pods are TLS encrypted. We choose haproxy because its so powerful and flexible configuration, great performance, low resource usage, and super good reliability.  

When haproxy does not have support to hot-update client certificate and CA, every time when the client and server TLS certificate are rotated, we have to reload entire haproxy (by whatever way: SIGHUP or master socket's `reload` command). We have some http/2 based GRPC stream calls between different pods, and reload to haproxy always left some old process still keeping the connection and they can not be dropped. If haproxy can have support of hot-updating client certificate and CA file, we don't have to reload haproxy just in order to rotate the service mesh TLS certificate for each pod.

**Comment by wlallemand** [comment]
Indeed the current "set ssl cert" command only support the crt part of the frontend side. It could be extended to the backend side now that everything is in place in the frontend.

**Comment by wlallemand** [comment]
Hot update for server certificates (crt keyword) was implemented for the 2.4 version. Development for the 2.5 version has started and will include ca-file and crl-file hot update (frontend/backend).

**Comment by wlallemand** [comment]
This was merged in 2.4 for "crt" and 2.5 for "ca-file/crl-file".
